### ISSUE TITLE
JavaScript: es6 module generation

### ISSUE THREAD

**Issue by tbillington** [google/protobuf#4274]
Asking because I couldn't find anything about generating es6 modules instead of closure or common via protoc (https://developers.google.com/protocol-buffers/docs/reference/javascript-generated#package).

Is it possible or on the roadmap?

**Comment by lfmunoz** [comment]
Yes it isn't specified anywhere what would be the correct syntax. My guess is this:
protoc --js_out=import_style=es6,binary:${DEST} -I ${SRC} ${PROTO_FILE}

but the code generated isn't showing any imports or exports so it must not be working. Can someone confirm what the correct syntax is?

**Comment by jwall149** [comment]
```

**Comment by tbillington** [comment]
@jwall149 Are you sure `random` isn't a valid option?

**Comment by xfxyjwf** [comment]
es6 module isn't supported yet.

**Comment by juanjoDiaz** [comment]
Any chance of ES6 imports happening soon?

Currently blocking https://github.com/grpc/grpc-web/issues/237

**Comment by sikmike** [comment]
Are there any news about ES6 support? I am trying to use grpc-web with commonjs in Vue.js, but get an error.

**Comment by thesayyn** [comment]
You should consider using https://github.com/thesayyn/protoc-gen-ts. It is generates classes ES6 compatible.

**Comment by seishun** [comment]
Would you accept ES6 support in a pull request?

**Comment by Asher-** [comment]
Why is it considered to be in any way acceptable to still be relying on ES5 code 5 years after its deprecation?

**Comment by ppeou** [comment]
Please enhance protoc compiler to generate es6 modules.

**Comment by gunters63** [comment]
I made a patch for an older commit of protoc for creating ES6 compatible code. It worked fine for me, although it is far from being good enough for a pull request:

https://github.com/protobufjs/protobuf.js/issues/1567
